Firestore/Source/API/FIRFirestore+Internal.h

```diff
@@ -34,7 +34,8 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)initWithProjectID:(NSString *)projectID
                          database:(NSString *)database
                    persistenceKey:(NSString *)persistenceKey
-              credentialsProvider:(id<FSTCredentialsProvider>)credentialsProvider
+              credentialsProvider:(firebase::firestore::auth::CredentialsProvider *)
```

Comment:
See comment in FSTIntegrationTestCase.mm. tl;dr: Consider a unique_ptr<> (assuming that's possible.)